frontend/catalyst/compilation_pipelines.py

```diff
@@ -324,10 +283,13 @@ def return_value_ptr_to_numpy(return_value_ptr):
         Returns:
             list or single numpy array
         """
-        ranked_memrefs = CompiledFunction.return_value_ptr_to_ranked_memrefs(return_value_ptr)
-        return_value = CompiledFunction.ranked_memrefs_to_numpy(ranked_memrefs)
-        # TODO: Handle return types correctly. Tuple, lists of 1 item (?)
-        return return_value[0] if len(return_value) == 1 else return_value
+        return_value = return_value_ptr.contents
+        numpy_arrays = []
+        for memref in return_value:
+            memref_ptr = ctypes.pointer(memref)
+            numpy_array = CompiledFunction.ranked_memref_to_numpy(memref_ptr)
+            numpy_arrays.append(numpy_array)
+        return numpy_arrays[0] if len(numpy_arrays) == 1 else numpy_arrays
```

Comment:
This is more subtle than what might be imagined at first glance. We do not require users to input jax arrays. Numpy works as well.

```python
from catalyst import qjit
import pennylane as qml
import numpy

@qjit
@qml.qnode(qml.device("lightning.qubit", wires=1))
def f(x):
    qml.RX(x[0], wires=0)
    return qml.state()

print(f(numpy.array([3.14])))
```

The reason why numpy works as an input is because it gets converted into a jax array. We could restrict the input to jax arrays or we could only require them internally and then match the return value with the expected type by the user.

Reply:
For UI/UX questions, I would always err on the side of 'how does JAX work'. In this case, you can always pass NumPy + JAX arrays to JAX functions (I believe?), but the output is always a `DeviceArray`. But would be good for someone to double check this!

Reply:
I just confirmed this locally. JAX allows anything `array_like` as input (`float`, `np.ndarray`, or `jax.Array`), and always returns `jax.Array`.

Reply:
Changed here https://github.com/PennyLaneAI/catalyst/pull/54/commits/c304d0c962b2c6f3358f35992a3a1cf32adb696a